### ISSUE TITLE
[Feature Request] Auto Update (master, dev, stable)

### ISSUE THREAD

**Issue by kathyrollo** [flyinghead/flycast#1091]
### Discussed in https://github.com/flyinghead/flycast/discussions/724

<div type='discussions-op-text'>

<sup>Originally posted by **djchadli** August 31, 2022</sup>
hello , I'm wondering why there is no update tool inside the app for windows , we will not miss any update once released . just a click to check and will know.</div>

Hi @flyinghead ,

Similar to other emulators where users can pick update frequency based on branch or update for stable releases only.
Thanks!

**Comment by flyinghead** [comment]
It looks like WinGet requires the app to have a windows installer, which flycast doesn't have.

**Comment by TreeBranches** [comment]
> It looks like WinGet requires the app to have a windows installer, which flycast doesn't have.

It has an APPX in the releases though right? That is supported.

Though a quick MSI creation with WiX https://wixtoolset.org/ could be scripted during the build process. I could assist with its creation if need be.

**Comment by flyinghead** [comment]
Yes, the UWP build is packaged as an APPX but it has limitations compared to the regular Windows version (directx11 only and more).

**Comment by kathyrollo** [comment]
Hi @flyinghead ,

This is what users tend to be familiar with these days. This is mostly from Qt-based emulators but anything that effectively does the same should be fine.

![Screenshot 2023-06-02 193607](https://github.com/flyinghead/flycast/assets/28589393/104200f4-a2ec-48f2-ab68-5297817a3ca1)

They can opt in/out of harassment. 😆

**Comment by vkedwardli** [comment]
We have an auto updater in our branch: https://github.com/inada-s/flycast/releases/tag/gdxsv-1.3.1
For those who are interested please try download the v1.3.1 and it should be able to auto update to v1.3.2

Currently it is a very simple updater which will always keep the older version in the same folder.

**Comment by mohsinous** [comment]
Hi @flyinghead 
plz use the updater for dev branch too
tnx

**Comment by davidhedlund** [comment]
The [AppImage Documentation - Making AppImages updateable](https://docs.appimage.org/packaging-guide/optional/updates.html) document has the information needed to properly support this.

quick-sharun is a wrapper over `lib4bin`/`sharun` that greatly simplifies deployment, it also offers additional features and fixes. ([Including an updater](https://github.com/pkgforge-dev/Anylinux-AppImages/blob/main/useful-tools/hooks/self-updater.hook))
